### PR TITLE
add `with drawing()` to public API

### PR DESCRIPTION
**Pull request by roberto-arista**
following this discussion: #474

### CODE REVIEW

File: drawBot/drawBotDrawingTools.py [hunk 1]
```diff
@@ -180,6 +199,27 @@ def endDrawing(self):
...
+            # and it uses the `with drawing()` statement to ensure that page numbers
+            # restart from 1 in each PDF
+            for eachBooklet in range(1, 4):
+                with drawing():
+                    for eachPage in range(10):
+                        newPage(400, 400)
+                        text(f"{pageCount()}", (100, 100))
+                    saveImage(f"book_{eachBooklet}.pdf")
+        """
+        return DrawingContextManager(self)
```
Comment (justvanrossum) [inline drawBot/drawBotDrawingTools.py:221]: Suggest to use a contextlib.contextmanager decorator on this function instead:

- https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager

Also, we'll need an "expected result" image for the demo code. I forgot what the easiest way to do that is.

**Comment by roberto-arista** [comment]
Hey Just, I've read the docs for `contextlib.contextmanager` but I am a bit confused on how it applies to this case. The docs say 

> The function being decorated must return a [generator](https://docs.python.org/3/glossary.html#term-generator)-iterator when called. This iterator must yield exactly one value, which will be bound to the targets in the [with](https://docs.python.org/3/reference/compound_stmts.html#with) statement’s as clause, if any.

How can `drawing()` return a generator-iterator?

**Comment by typemytype** [comment]
I will add the expected example test images

**Comment by justvanrossum** [comment]
As per the example from the docs, it should look roughly like this:

```python
    contextlib.contextmanager
    def drawing(self):
        self.newDrawing()
        try:
            yield
        finally:
            self.endDrawing()
```

**Comment by roberto-arista** [comment]
The code should be fine, now it seems I have to figure out the tests images

**Comment by typemytype** [comment]
the same contextmanager could be used for `savedState`

**Comment by roberto-arista** [comment]
👋

**Comment by roberto-arista** [comment]
Hey all!
Is there anything I can do to push forward this pull request?
I think it would be a nice addition to the drawbot API!

**Comment by roberto-arista** [comment]
🎉

**Comment by justvanrossum** [comment]
> I could not open a separate pull request because this one has not been merged yet.

PR's don't need to be merged in order of creation, especially if their respective changes don't conflict.

And once one is merged, it is possible to rebase the other.

**Comment by roberto-arista** [comment]
Wait, I thought this was merged, instead it was only approved. I'll reopen the pull request!

**Pull request merged**